src/main/scala/nl.knaw.dans.easy.bagstore/component/FileSystemComponent.scala

```diff
@@ -243,10 +240,10 @@ trait FileSystemComponent extends DebugEnhancedLogging {
           _ = debug(s"valid tagmanifests: $validTagManifests")
           _ <- removeFetchTxtAndTagManifests(workBag)
           _ <- createSymLinks(mappings)
-          (valid, msg) <- bagFacade.isValid(workBag)
-          _ = debug(s"valid bag: $valid")
+          validWorkBag <- bagFacade.isValid(workBag)
+          _ = debug(validWorkBag.fold(msg => s"invalid bag: $msg", _ => "valid bag"))
           _ <- Try { FileUtils.deleteDirectory(tempDir.toFile) }
-        } yield (validTagManifests && valid, msg)
+        } yield validWorkBag.fold(msg => Left(msg), _ => if (validTagManifests) Right(()) else Left("tagmanifests are not valid"))
```

Comment:
This was probably already the case, but I suspect the error message will *not* include a message about invalid tag manifests if the work bag was also invalid.